examples/plot_log_scale/src/main.rs

```diff
@@ -0,0 +1,152 @@
+        Box::new(|_cc| Ok(Box::<PlotExample>::default())),
+    )
+}
+
+struct PlotExample {
+    log_x: bool,
+    log_y: bool,
+    signals: Vec<Signal>,
+}
+
+struct Signal {
+    name: &'static str,
+    points: Vec<[f64; 2]>,
+}
+
+impl Default for PlotExample {
+    fn default() -> Self {
+        let x = (-2000..2000).map(|x| x as f64 / 100.0);
+        let signals = vec![
+            Signal {
+                name: "y=x",
+                points: x.clone().map(|x| [x, x]).collect(),
+            },
+            Signal {
+                name: "y=x^2",
+                points: x.clone().map(|x| [x, x.powi(2)]).collect(),
+            },
+            Signal {
+                name: "y=exp(x)",
+                points: x.clone().map(|x| [x, x.exp()]).collect(),
+            },
+            Signal {
+                name: "y=ln(x)",
+                points: x.clone().map(|x| [x, x.ln()]).collect(),
+            },
+        ];
+        Self {
+            log_x: true,
+            log_y: true,
+            signals,
+        }
+    }
+}
+
+impl eframe::App for PlotExample {
+    fn update(&mut self, ctx: &egui::Context, _: &mut eframe::Frame) {
+        egui::SidePanel::left("options").show(ctx, |ui| {
+            ui.checkbox(&mut self.log_x, "X axis log scale");
+            ui.checkbox(&mut self.log_y, "Y axis log scale");
+        });
+        let log_x = self.log_x;
+        let log_y = self.log_y;
+        egui::CentralPanel::default().show(ctx, |ui| {
+            let mut plot = egui_plot::Plot::new("plot")
+                .legend(Legend::default())
+                .label_formatter(|name, value| {
+                    let x = if log_x {
+                        10.0f64.powf(value.x)
+                    } else {
+                        value.x
+                    };
+                    let y = if log_y {
+                        10.0f64.powf(value.y)
+                    } else {
+                        value.y
+                    };
+                    if !name.is_empty() {
+                        format!("{name}: {x:.3}, {y:.3}")
+                    } else {
+                        format!("{x:.3}, {y:.3}")
+                    }
+                });
+            if log_x {
+                plot = plot
+                    .x_grid_spacer(log_axis_spacer)
+                    .x_axis_formatter(log_axis_formatter);
+            }
+            if log_y {
+                plot = plot
+                    .y_grid_spacer(log_axis_spacer)
+                    .y_axis_formatter(log_axis_formatter);
+            }
+            plot.show(ui, |plot_ui| {
+                for signal in &self.signals {
+                    let points: Vec<_> = signal
+                        .points
+                        .iter()
+                        .copied()
+                        .map(|[x, y]| {
+                            let x = if log_x { x.log10() } else { x };
+                            let y = if log_y { y.log10() } else { y };
```

Comment:
when x or y is negative, `log10` is not valid and returns `NaN`

Reply:
Yes, normally you can't represent non-positive numbers on a log scale. Apparently there is a modified transform that allows that, but I think this starts to get too complicated for this example.

It is also noteworthy that the infinity value produced when we take the log of 0 causes the autobounds to stop working (it does not seem to be bothered by the NaNs though), so that is something that could be looked into, but in a separate PR.

Reply:
Yeah, thx, I am basically trying to do a bode plot, I will try to understand the example a bit and maybe do the transform on my end
https://en.wikipedia.org/wiki/Bode_plot

Reply:
Yeah, for a Bode plot the y axis is in dB, so you don't need the log scale for it.